Frontend/CarRentalAdmin/src/app/login/login.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { HttpErrorResponse } from '@angular/common/http';
+
+import { LoginRequest } from '../_models/login-request';
+
+import { AuthService } from '../_services/auth.service';
+
+@Component({
+    selector: 'app-login',
+    templateUrl: './login.component.html',
+    styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+    public adminCredentials: LoginRequest = {
+        username: '',
+        password: ''
+    };
+
+    public authFailed: boolean = false;
+    public authFailedMessage: string = '';
+    private isVisiblePassword: boolean = false;
+    private returnUrl: string = '';
+
+
+    constructor(
+        private authService: AuthService,
+        private route: ActivatedRoute,
+        private router: Router
+    ) {
+    }
+
+    public ngOnInit(): void {
+        this.returnUrl = this.route.snapshot.queryParams['returnUrl'] || '';
+    }
+
+    public resetAuthStatus(): void {
+        this.authFailed = false;
+        this.authFailedMessage = '';
+    }
+
+    public changePasswordType(): void {
+        this.isVisiblePassword = !this.isVisiblePassword;
+    }
+
+    public getPasswordType(): string {
+        return this.isVisiblePassword ? 'text' : 'password';
+    }
+
+    public getButtonText(): string {
+        return this.isVisiblePassword ? 'Hide' : 'Show';
+    }
+
+    public onSubmit(): void {
+        this.authService.login(this.adminCredentials)
+            .subscribe(
+                _ => {
+                    this.router.navigateByUrl(this.returnUrl);
+                },
+                error => {
+                    this.handleError(error);
+                    this.authFailed = true;
+                }
+            );
+    }
+
+    private handleError(error: HttpErrorResponse): void {
```

Comment:
looks like you will need notification service to handle user-friendly messages
I suggest to create separate task if you don't want to do it inside current task